### ISSUE TITLE
Need the ability to look up hardware vendors by ID

### ISSUE THREAD

**Issue by sungo** [joyent/conch#596]
API v2.22, specifically #519, changed the HardwareProduct response to send a UUID for the vendor instead of its string name. Near as I can tell, there is no way to look up a hardware vendor by its UUID. Specifically https://github.com/joyent/conch/blob/v2.22.0/lib/Conch/Route/HardwareVendor.pm#L34 and https://github.com/joyent/conch/blob/v2.22.0/lib/Conch/Controller/HardwareVendor.pm#L26 only search on the name column.

When displaying a hardware product, the shell shows the vendor name. As of v2.22, I can no longer do that. I either need the 'vendor' field restored on HardwareProduct or I need an endpoint to look up a HardwareVendor by its UUID.

**Comment by karenetheridge** [comment]
Yup, looks like i should have fixed Conch::Controller::HardwareVendor::find_hardware_vendor to look up by id instead of (or as well as?) by name.

I can just do a `is_uuid($path_value)` check for now to support both for now, and remove the name lookup later if not needed.

**Comment by sungo** [comment]
"as well as" please. I have quite a few hardware vendor functions that use the string name for lookups. I'd rather not modify all of those to handle a regression.
